web/src/main/java/org/openmrs/web/controller/encounter/EncounterDisplayController.java

```diff
@@ -387,6 +387,25 @@ public int compareTo(FieldHolder other) {
 		}
 		
 		/**
+		 * Indicates whether some other object is "equal to" this one.
+		 *
+		 * @see java.lang.Object#equals(java.lang.Object)
+		 */
+		public boolean equals(FieldHolder other) {
+			return compareTo(other) == 0;
+		} //
+		
+		/**
+		 * Returns a hash code based on the formField fields used in comparing.
+		 * @see java.lang.Object#hashCode()
+		 */
+		public int hashCode() {
+			return formField.getPageNumber() * formField.getFieldNumber()
```

Comment:
The rule is that hashCodes must be the same if equals returns true. It's not the case here if pageNumber is equal and some other field is different. The easiest fix here is to include just the first field that is checked in compareTo i.e. pageNumber. Please use HashCodeBuilder to generate the value.